Aquib/1_Valid_Sudoku.c.txt

```diff
@@ -0,0 +1,83 @@
+bool checkRow(char **board, int n)
+{
+    char boardstatus[10] = {0};
+    for(int i =0; i < 9;i++)
+    {
+         char cell = board[n][i];
+        if(board[n][i] != '.')
+        {
+            char idx = cell - '0';
+            if(boardstatus[idx] == 0)
+            {
+                boardstatus[idx] = cell;
+            }
+            else
+            {
+                return false;
+            }
+        }
+    }
+    return true;
+}
+
+bool checkCol(char **board, int n)
+{
+    char boardstatus[10] = {0};
+    for(int i =0; i < 9;i++)
+    {
+         char cell = board[i][n];
+        if(board[i][n] != '.')
+        {
+            char idx = cell - '0';
+            if(boardstatus[idx] == 0)
+            {
+                boardstatus[idx] = cell;
+            }
+            else
+            {
+                return false;
+            }
+        }
+    }
+    return true;
+    
+}
+
+bool checkGrid(char **board, int n)
+{
+    char boardstatus[10] = {0};
+    int row = (n /3) * 3;
+    int col = (n % 3) * 3;
+    for(int r = row; r < (row+3);r++)
+    {
+        for(int c = col; c < (col+3);c++)
+        {
+            char cell = board[r][c];
+            if(board[r][c] != '.')
+            {
+              char idx = cell - '0';
+              if(boardstatus[idx] == 0)
+              {
+                boardstatus[idx] = cell;
+              }
+              else
+              {
+                return false;
+              }
+        }
+        }
+    }
+    return true;    
+}
+
+
+
+bool isValidSudoku(char** board, int boardSize, int* boardColSize)
```

Comment:
Where is boardColSize used? Is it really needed?

Reply:
It is not used in this case. However, in C it is a practise to allow user of that function to pass an address in memory location to get the size of any input array that might not be clear. I can assign any value to that variable by dereferencing it (by doing *boardColSize = 9;). I haven't provided any return value to it because in this case that variable is useless. It came with the template of the function that leetcode provided, so I didn't touch it. I thought it may disrupt the compilation as per their scripts.

Reply:
Right, or you could just pass it to checkCol instead of keeping it unused since it may come across as misleading. Check if this helps https://rules.sonarsource.com/c/RSPEC-1172/#:~:text=%C2%A0Code%20Smell-,Unused%20function%20parameters%20should%20be%20removed,-Analyze%20your%20code
No need to make any changes, you can keep that in mind in the actual interview